syntax="proto2";

package QRadioLink;

option optimize_for = SPEED;

message PageMessage {
    required string calling_user = 1;
    required string called_user = 2;
    optional bool retransmit = 3;
    optional string via_node = 4;
}

message Channel {
    optional uint32 channel_id = 1;
    optional uint32 parent_id = 2;
    optional string name = 3;
    optional string description = 4;
}

message User {
    optional uint32 session = 1;
    optional string name = 2;
    optional uint32 user_id = 3;
    optional uint32 channel_id = 4;
    optional string comment = 5;
}

message RepeaterInfo {
    message Channel {
        optional uint32 channel_id = 1;
        optional uint32 parent_id = 2;
        optional string name = 3;
        optional string description = 4;
    }
    message User {
        optional uint32 session = 1;
        optional string name = 2;
        optional uint32 user_id = 3;
        optional uint32 channel_id = 4;
        optional string comment = 5;
    }
    repeated Channel channels = 1;
    repeated User users = 2;
}

message Parameters {
    required uint32 station_id = 1;
    optional uint32 in_call = 2;
    optional uint32 channel_id = 3;
    optional uint32 caller_id = 4;
}

message JoinConference {
    required uint32 server_id = 1;
    required uint32 channel_id = 2;
    required uint32 caller_id = 3;
    optional string server_ip = 4;
    optional uint32 server_port = 5;
}
    
message LeaveConference {
    required bool leave = 1;
}
